src/compiler/checker.ts

```diff
@@ -18514,10 +18514,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return !!(getUnionType([intersectTypes(type1, type2), neverType]).flags & TypeFlags.Never);
     }
 
+    function hasOptionalModifiersProperty(mappedType: MappedType) {
+        const modifiersType = getApparentType(getModifiersTypeFromMappedType(mappedType));
+        if (!(modifiersType.flags & TypeFlags.Object)) {

```

Comment:
TODO: might also have to handle intersection (that's what `isWeakType` is doing) and I need to recheck behavior with unions

Reply:
recheck behavior with union types later

Reply:
@Mateusnasciment Please try to keep comments actionable and on-topic